src/sources/soundsourceproxy.cpp

```diff
@@ -464,12 +464,11 @@ void SoundSourceProxy::loadTrackMetadataAndCoverArt(
         parsedFromFile = true;
         if (!coverImg.isNull()) {
             // Cover image has been parsed from the file
-            coverArt.image = coverImg;
             // TODO() here we may introduce a duplicate hash code
-            coverArt.info.hash = CoverArtUtils::calculateHash(coverArt.image);
-            coverArt.info.coverLocation = QString();
-            coverArt.info.type = CoverInfo::METADATA;
-            coverArt.info.source = CoverInfo::GUESSED;
+            coverInfoRelative.hash = CoverArtUtils::calculateHash(coverImg);
```

Comment:
Yes, I have thought the same. I have not done it, because the I like to verify my final cover art duplicate fix first. Some changes that are here almost natural, but  will not work in the final branch.